modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc

```diff
@@ -334,78 +351,74 @@ class NDB_Menu_Filter_Mri_Upload extends NDB_Menu_Filter_Form
+                ///////////////////////////////////////////////////////////////////
+                $visit_label = $args['values']['VisitLabel'];
+                $candid      = $args['values']['CandID'];
+                $pscid       = $args['values']['PSCID'];
+                $pname       = $pscid. "_" . $candid . "_" . $visit_label;
+                $sessionid   = $db->pselectOne(
+                    "SELECT ID FROM  session WHERE CandID = :cid
+                     AND Visit_label =:vlabel",
+                    array(
+                     'cid'    => $candid,
+                     'vlabel' => $visit_label,
+                    )
                 );
             }
-
-            return true;
         }
-        return false;
+        //////////////////////////////////////////////////////////////////
+        ///Get the directory to which the temp file has been moved to/////
+        /// MRI_upload Incoming directory/////////////////////////////////
+        //////////////////////////////////////////////////////////////////
+        $uploaded_file_path = trim(
+            $file->getDestinationDirectory() .
+            $file->getDestinationFilename()
+        );
+        ///////////////////////////////////////////////////////////////
+        ///if the DestinationDirectory/getDestinationFilename//////////
+        //is not empty, then remove the uploaded directory in the /tmp/
+        ////Otherwise keep uploaded directory in the /tmp and insert///
+        ////into the mri_upload table//////////////////////////////////
+        ///////////////////////////////////////////////////////////////
+        if ($uploaded_file_path) {
+            $this->removeTempDir();
+        } else {
+                $uploaded_file_path = $this->getTempPath();
+        }
+        $values = array(
+                   'UploadedBy'     => $user_name,
+                   'UploadDate'     => $date,
+                   'UploadLocation' => $uploaded_file_path,
+                   'SessionID'      => $sessionid,
+                   'PatientName'    => $pname,
+                   'IsPhantom'      => $IsPhantom,
+                  );
+        $db->Insert(
+            'mri_upload',
+            $values
+        );
+        // Get MRI upload ID of the upload entry that was just inserted
+        $query       = "SELECT UploadID
+                       FROM mri_upload
+                       WHERE UploadedBy     = :UploadedBy
+                       AND   UploadDate     = :UploadDate
+                       AND   UploadLocation = :UploadLocation
+                       AND   SessionID      = :SessionID
+                       AND   PatientName    = :PatientName
+                       AND   IsPhantom      = :IsPhantom";
+        $mriUploadId = $db->pselectOne($query, $values);
+        // Perform the real upload on the server
+        $serverProcessLauncher = new ServerProcessLauncher();
```

Comment:
Should you check the auto launch config setting here before calling the launcher?